include/ed/models/model_loader.h

```diff
@@ -22,15 +22,63 @@ class ModelLoader
+     * are considered. Otherwise only ED yaml models are used.
+     * @param id new id of the entity
+     * @param type type of the entity
+     * @param req UpdateRequest to 'fill'
+     * @param error error stream
+     * @param allow_sdf Allow SDF models, or only ED yaml models
+     * @return bool, which indicates succes
+     */
+    bool create(const UUID& id, const std::string& type, UpdateRequest& req, std::stringstream& error, const bool allow_sdf=false);
+
+    /**
+     * @brief create add entity to update_request from config data. "_root" will be used as id.
+     * @param data should contain the right data and should be at the correct reading point
+     * @param req UpdateRequest to 'fill'
+     * @param error error stream
+     * @return bool, which indicates succes
+     */
     bool create(const tue::config::DataConstPointer& data, UpdateRequest& req, std::stringstream& error);
 
+    /**
+     * @brief create add entity to update_request from config data.
+     * @param data should contain the right data and should be at the correct reading point
+     * @param id_opt Optional id, if no id in data
+     * @param parent_id prefix to id, unless starting with "_"
+     * @param req UpdateRequest to 'fill'
+     * @param error error stream
+     * @param model_path path where the data is read. Is used for relative paths in the data
+     * @param pose_offset pose offset, if no pose in data, this pose will be the final pose.
+     * @return bool, which indicates succes
+     */
     bool create(const tue::config::DataConstPointer& data, const UUID& id_opt, const UUID& parent_id,
                 UpdateRequest& req, std::stringstream& error, const std::string& model_path = "",
                 const geo::Pose3D& pose_offset = geo::Pose3D::identity());
 
-    bool exists(const std::string& type) const;
+    /**
+     * @brief createSDF add SDF entity to update_request from config data.
+     * @param data should contain the right data and should be at the correct reading point
+     * @param parent_id prefix to id, unless starting with "_"
+     * @param parent_pose pose offset which will always be applied
+     * @param id_override if not empty, this will override the id of the model
+     * @param pose_override if not empty, this will override the pose of the model
+     * @param req UpdateRequest to 'fill'
+     * @param error error stream
+     * @return bool, which indicates succes
+     */
+    bool createSDF(const tue::config::DataConstPointer& data, const UUID& parent_id, const geo::Pose3D& parent_pose,
+                   const UUID& id_override, const boost::shared_ptr<const geo::Pose3D> pose_override,
+                   UpdateRequest& req, std::stringstream& error);
+
+    /**
+     * @brief exists Check of a model of type 'type' exist
+     * @param type model type
+     * @return bool, which indicates of type exist
+     */
+    bool exists(std::string type) const;
```

Comment:
I'd prefer `has_model_of_type`, or at least something more precise than `exists`, as lots of things exist in a loader. Not sure `ModelLoader` should have this, as a loader loads stuff rather than store it.

Reply:
This functionality already existed XD before this PR. So out of scope of this PR.